src/settings.rs

```diff
@@ -79,6 +79,9 @@ impl tracing_subscriber::fmt::time::FormatTime for TimeFormatter {
 /// Settings for proof-of-coverage (PoC).
 #[derive(Debug, Deserialize, Clone)]
 pub struct PocSettings {
+    // Enable/disable poc related activities (baecon/witness)
+    #[serde(default = "default_poc_enable")]
+    pub enable: bool,
```

Comment:
why not rename this to disable? you would avoid this default override plus the only if block that uses this boolean (`handle_received_beacon`) inverts is anyway.

Reply:
Oh nevermind, it's used elsewhere too. Not a big deal either way